dashboard/app/controllers/api/v1/pd/application/teacher_applications_controller.rb

```diff
@@ -11,12 +11,18 @@ def new_form
       )
     end
 
-    # PATCH /api/v1/pd/application/teacher/<applicationId>
+    # PUT /api/v1/pd/application/teacher/<applicationId>
     def update
-      form_data_hash = params.try(:[], :form_data) || {}
-      form_data_json = form_data_hash.to_unsafe_h.to_json.strip_utf8mb4
+      form_data_hash = params.try(:[], :form_data)
+      new_status = params.try(:[], :status)
+      return unless form_data_hash || new_status
```

Comment:
I'm not sure about other existing patterns, but my instinct would be to either create a new endpoint like `update_status` or in the client code, send the whole form_data_hash to update the status, instead of overloading `update` to handle two different cases depending on what gets sent in. For example, what happens if form_data and status get sent to this api? I think it just updates the status and ignores whats in form_data